students/kmsnyde/lesson06/test_mailroom4.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Apr 28 13:19:54 2018
+
+@author: Karl M. Snyder
+"""
+
+"""
+Define tests that will work againsts mailroom4.py.
+"""
+
+import mailroom4 as mail
+import os.path
+
+def test_1():
+    donors = mail.donor_dict
+    assert "Karl Stick" in donors
+    
+#IPython will display "Donor Amount:", input same as below for pass, or diff
+# to fail
+def test_2():
+    assert mail.donations() == 50.00
+
+
+#In IPython promp after Donor Amount: from above test, type "May West" and then any number.
+#If you type May West, no error; else AssertionError
+def test_3():
+    mail.send_thanks()
+    assert "May West" in mail.donor_dict
```

Comment:
Your tests should be able to be fully automated, which means they should be able to run without any external inputs. That makes testing some parts of mailroom a bit tricky. The best approach is to separate the parts of your program that collect user input from the business logic and to write small, focused functions with testable return values based on various forms of input